Recipes/Recipes.DataAccess/Interfaces/IGenericRepository.cs

```diff
@@ -0,0 +1,10 @@
+﻿namespace Recipes.DataAccess.Interfaces;
+
+public interface IGenericRepository<T> where T : class
+{
+    public Task<T> CreateAsync(T entity);
+    public Task<T> GetByIdAsync(int id);
+    public Task<T> GetAllAsync();
+    public Task<T> DeleteByIdAsync(int id);
```

Comment:
You probably wouldn't return the no longer existant object.

```suggestion
    public Task DeleteByIdAsync(int id);
```